src/main/java/net/minecraftforge/common/config/Configuration.java

```diff
@@ -789,240 +808,212 @@ public boolean hasKey(String category, String key)
         return cat != null && cat.containsKey(key);
     }
 
-    public void load()
+    public void load(BufferedReader buffer)
     {
-        if (PARENT != null && PARENT != this)
-        {
-            return;
-        }
-
-        BufferedReader buffer = null;
-        UnicodeInputStreamReader input = null;
         try
         {
-            if (file.getParentFile() != null)
-            {
-                file.getParentFile().mkdirs();
-            }
-
-            if (!file.exists())
+            String line;
+            ConfigCategory currentCat = null;
+            Property.Type type = null;
+            ArrayList<String> tmpList = null;
+            int lineNum = 0;
+            String name = null;
+            loadedConfigVersion = null;
+
+            while (true)
             {
-                // Either a previous load attempt failed or the file is new; clear maps
-                categories.clear();
-                children.clear();
-                if (!file.createNewFile())
-                    return;
-            }
+                lineNum++;
+                line = buffer.readLine();
 
-            if (file.canRead())
-            {
-                input = new UnicodeInputStreamReader(new FileInputStream(file), defaultEncoding);
-                defaultEncoding = input.getEncoding();
-                buffer = new BufferedReader(input);
-
-                String line;
-                ConfigCategory currentCat = null;
-                Property.Type type = null;
-                ArrayList<String> tmpList = null;
-                int lineNum = 0;
-                String name = null;
-                loadedConfigVersion = null;
-
-                while (true)
+                if (line == null)
                 {
-                    lineNum++;
-                    line = buffer.readLine();
+                    if (lineNum == 1)
```

Comment:
Missing brackets around many conditionals, missing space before the brackets of the empty constructor....